packages/@react-aria/tooltip/src/useTooltipTrigger.ts

```diff
@@ -25,57 +26,90 @@ interface TooltipTriggerAria {
 
 export function useTooltipTrigger(props: TooltipTriggerAriaProps, state: TooltipTriggerState, ref: RefObject<HTMLElement>) : TooltipTriggerAria {
   let {
-    tooltipProps = {} as TooltipProps,
-    triggerProps = {} as TriggerProps,
     isDisabled
   } = props;
 
-  let tooltipId = useId(tooltipProps.id);
-  let triggerId = useId(triggerProps.id);
+  let tooltipId = useId();
+
+  let isHovered = useRef(false);
+  let isFocused = useRef(false);
+
+  let handleShow = () => {
+    if (isHovered.current || isFocused.current) {
+      state.open(isFocused.current);
+    }
+  };
+
+  let handleHide = () => {
+    if (!isHovered.current && !isFocused.current) {
+      state.close();
+    }
+  };
 
   let onKeyDownTrigger = (e) => {
```

Comment:
should this really be on the document body?

Reply:
mm, is this so you don't have to be focused on the button in order for esc to close the tooltip? That sounds legit

Reply:
yes, I think it should be a global listener.